lib/ansible/modules/cloud/amazon/redshift_iam_roles.py

```diff
@@ -0,0 +1,140 @@
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+module: redshift_iam_roles
+short_description: This module adds/removes IAM roles to/from Redshift clusters.
+version_added: 2.6
+author: Aaron Smith (@slapula)
+description:
+    - "This module adds or removes a list of IAM roles to a give Redshift Cluster"
+options:
+    cluster:
+        description:
+            - "This is the cluster you'd like to modify"
+        required: true
+    state:
+        description:
+            - "Desired state of the list of IAM roles on the give Redshift cluster"
+        required: true
+        choices: ['present', 'absent']
+    roles:
+        description:
+            - "This is a list of IAM roles (Full ARN) to add to the Redshift cluster. Roles attached to the cluster that are not on the list will be removed."
+        required: true
+extends_documentation_fragment:
+    - ec2
+    - aws
+'''
+
+EXAMPLES = '''
+- name: add IAM roles to a redshift cluster
+  redshift_iam_roles:
+    cluster: "staging"
+    state: present
+    roles:
+        - "arn:aws:iam::123456789012:role/superuser"
+        - "arn:aws:iam::123456789012:role/new_hotness"
+
+- name: remove all IAM roles from a redshift cluster
+  redshift_iam_roles:
+    cluster: "staging"
+    state: absent
+'''
+
+RETURN = ''' # '''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import boto3_conn, ec2_argument_spec, get_aws_connection_info, camel_dict_to_snake_dict, HAS_BOTO3
+import traceback
+
+try:
+    from botocore.exceptions import ClientError, BotoCoreError
+except ImportError:
+    pass  # will be detected by imported HAS_BOTO3
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            cluster=dict(type='str', required=True),
+            state=dict(type='str', required=True, choices=['present', 'absent']),
+            roles=dict(type='list', required=True)
+        )
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True
+    )
+
+    if not HAS_BOTO3:
+        module.fail_json(msg='boto3 required for this module')
+
+    target_cluster = module.params.get('cluster')
+    desired_state = module.params.get('state')
+    target_roles = module.params.get('roles')
+
+    result = {
+        'changed': False
+    }
+
+    region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+    conn = boto3_conn(module, conn_type='client', resource='redshift', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+
+    cluster_roles = conn.describe_clusters(ClusterIdentifier=target_cluster)['Clusters'][0]['IamRoles']
+    current_roles = [x['IamRoleArn'] for x in cluster_roles]
+
+    if desired_state == 'present':
+        roles_to_add = list(set(target_roles) - set(current_roles))
+        roles_to_remove = list(set(current_roles) - set(target_roles))
```

Comment:
I think it might be good to have a `purge_roles` option to be able to only enable role_to_remove if desired. I think it's a little confusing that both state: absent and state: present can be used to remove roles. I like the logic in this section (particularly if there was a purge_roles option that works like purge_tags does for other AWS Ansible modules) better than for state absent.

Reply:
My goal here was to create a desired list of IAM roles that should be attached to a given cluster, apply those roles, and remove any that are not on the list.  That way you can have an authoritative task for setting Redshift IAM roles instead of placing/pruning specific roles.